lib/MqttClient.js

```diff
@@ -29,26 +29,19 @@ const HA_STATES = {
     IDLE: "idle",
     RETURNING: "returning",
     DOCKED: "docked",
-    ERROR: "error",
-    ZONE_CLEANUP: "cleaning"
+    ERROR: "error"
 };
 
-// Codes as per Status.js
-const HA_STATE_MAPPINGS = {
-    "CHARGER_DISCONNECTED": HA_STATES.IDLE,
-    "IDLE": HA_STATES.IDLE,
-    "CLEANING": HA_STATES.CLEANING,
-    "MANUAL_MODE": HA_STATES.CLEANING,
-    "SPOT_CLEANING": HA_STATES.CLEANING,
-    "GOING_TO_TARGET": HA_STATES.CLEANING,
-    "ZONED_CLEANING": HA_STATES.ZONE_CLEANUP,
-    "RETURNING_HOME": HA_STATES.RETURNING,
-    "DOCKING": HA_STATES.RETURNING,
-    "CHARGING": HA_STATES.DOCKED,
-    "CHARGING_PROBLEM": HA_STATES.ERROR,
-    "ERROR": HA_STATES.ERROR,
-    "PAUSED": HA_STATES.PAUSED,
-};
+
+const HA_STATE_MAPPINGS = {};
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.ERROR] = HA_STATES.ERROR;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.DOCKED] = HA_STATES.DOCKED;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.IDLE] = HA_STATES.IDLE;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.RETURNING] = HA_STATES.RETURNING;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.CLEANING] = HA_STATES.CLEANING;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.PAUSED] = HA_STATES.PAUSED;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.MANUAL_CONTROL] = HA_STATES.CLEANING;
+HA_STATE_MAPPINGS[attributes.StatusAttribute.VALUE.MOVING] = HA_STATES.CLEANING;
```

Comment:
```suggestion
const HA_STATE_MAPPINGS = {
    [attributes.StatusAttribute.VALUE.ERROR]: HA_STATES.ERROR,
    [attributes.StatusAttribute.VALUE.DOCKED]: HA_STATES.DOCKED,
    [attributes.StatusAttribute.VALUE.IDLE]: HA_STATES.IDLE,
    [attributes.StatusAttribute.VALUE.RETURNING]: HA_STATES.RETURNING,
    [attributes.StatusAttribute.VALUE.CLEANING]: HA_STATES.CLEANING,
    [attributes.StatusAttribute.VALUE.PAUSED]: HA_STATES.PAUSED,
    [attributes.StatusAttribute.VALUE.MANUAL_CONTROL]: HA_STATES.CLEANING,
    [attributes.StatusAttribute.VALUE.MOVING]: HA_STATES.CLEANING
};
```

Reply:
Oh!

I didn't know that this notation exists. Thanks!